### DISCUSSION TITLE
Backend parser - Summarise / Summarize / Group By

### DISCUSSION THREAD

**Discussion by yesoreyeram** [grafana/grafana-infinity-datasource#832] [Docs]
When you use backend parser, you can use `Summarize` option to group the results together. Examples given below

## Without summarize

<img width="1716" alt="image" src="https://github.com/grafana/grafana-infinity-datasource/assets/153843/4e3cdc35-dbad-49ae-8447-9b935dfb90d5">

## With summarize

<img width="1728" alt="image" src="https://github.com/grafana/grafana-infinity-datasource/assets/153843/979fce87-d4c4-483c-80a1-03f99a1c7705">

## With summarize by

<img width="1726" alt="image" src="https://github.com/grafana/grafana-infinity-datasource/assets/153843/d7e5976a-ce11-429c-8597-ed8a5c00cd9c">

## With summarize alias

<img width="1718" alt="image" src="https://github.com/grafana/grafana-infinity-datasource/assets/153843/ec498917-9e22-4642-8969-d392b4eba10d">

**Comment by yesoreyeram** [comment]
## Summarize Alias

When you have multiple queries with summarize option, all the values will be shown as `summary`. This makes hard to differentiate the values and also hard to operate when used with grafana merge transformation.


### Example: Without alias and merge transformation

Here I am using two different queries. One for returning mean age and another one for mean salary  

<img width="1277" alt="image" src="https://github.com/grafana/grafana-infinity-datasource/assets/153843/534cf8c4-6d23-47e2-83bc-0ed1ef6080fa">
<img width="1304" alt="image" src="https://github.com/grafana/grafana-infinity-datasource/assets/153843/e4d552d2-bc54-408b-a40c-6a722e6b35ce">
<img width="1718" alt="image" src="https://github.com/grafana/grafana-infinity-datasource/assets/153843/807a027b-b7b4-4521-ad99-a52e462c9fe0">

then I am using merge transformation to merge these two results.

<img width="1717" alt="image" src="https://github.com/grafana/grafana-infinity-datasource/assets/153843/6e237e13-f6c1-4a87-b1fb-55cb11d83ce1">

but this is not what I want also Join by field transformation also doesn't give me the result I want.

<img width="1722" alt="image" src="https://github.com/grafana/grafana-infinity-datasource/assets/153843/c50bbc0c-dc6d-4399-bfac-1977dc6c3407">

## Solution

The solution here is to use summarize alias in the queries to differentiate the metrics.

<img width="1029" alt="image" src="https://github.com/grafana/grafana-infinity-datasource/assets/153843/cc7207ff-cf70-4c91-821f-0d61323dda8d">

<img width="977" alt="image" src="https://github.com/grafana/grafana-infinity-datasource/assets/153843/587a8da4-2f4d-45a9-ac2c-9fecd622f10c">

then just using the merge transformation should work 

<img width="1710" alt="image" src="https://github.com/grafana/grafana-infinity-datasource/assets/153843/fc363fab-9a1d-4b4a-895e-01e471c8ec2e">

**Comment by preshit1505** [comment]
Hello @yesoreyeram, 
I am having some difficulty using this plugin. So my csv file is in GCS bucket which finally i was able to get the connection. 
This is how my CSV file looks like:
Secret, Namespace, Expiration Date, Days Remaining
argo-server-tls, argocd, Jun 25 13:16:45 2024 GMT,59
deployment-orchestratortls, bedrock-cli, Jun 21 11:52:35 2024 GMT,55
oauth2-tls, bedrock-portal, Jun 26 20:40:53 2024 GMT,60
portaltls, bedrock-portal, Jun 25 13:07:43 2024 GMT,59
bedrock-ford-selfsigned-ca, cert-manager, Nov  2 08:42:34 2033 GMT,3476
cvc-bedrock-jagent-user, cvc-bedrock-jagent, Nov  8 19:45:09 2024 GMT,195
bootstrap-service-gateway-tls, cvc-bootstrap-service, Nov  7 04:17:21 2023 GMT,-171
cvc-data-dist-catalog-m-user, cvc-data-dist-catalog, Apr 11 00:19:57 2025 GMT,349
cvc-lakehouse-database-user, cvc-lakehouse, Jan 30 21:22:06 2025 GMT,278
cvc-lakehouse-user, cvc-lakehouse, Nov  6 19:05:15 2024 GMT,193
cvc-mw-scratchpad-tls, cvc-mw-scratchpad, Jul 24 21:58:16 2024 GMT,89


**Issue Facing:**
<img width="1130" alt="image" src="https://github.com/grafana/grafana-infinity-datasource/assets/14802056/505f8a9d-adfd-4be8-8cce-107eaf0a8579">
I am trying this put this configuration but this is how i am seeing my data (also i tried to add the column names and type)

<img width="1413" alt="image" src="https://github.com/grafana/grafana-infinity-datasource/assets/14802056/91f0520f-bab9-4156-8f59-78d1319605cb">


Also, I tried using UQL parser but it didnt work too
![image](https://github.com/grafana/grafana-infinity-datasource/assets/14802056/b2855e8c-2afd-4205-bad1-4ae1fdaf0b23)

Can you please help me in getting the correct format or configuration to get the data?

**Reply by yesoreyeram** [reply]
the response you are getting seems to be some html response (signin page) rather than the actual CSV. So I doubt that there is an issue with your datasource config.
